src/Model.elm

```diff
@@ -0,0 +1,54 @@
+module Model exposing (..)
+
+import Dict exposing (Dict)
+import RemoteData exposing (RemoteData(..))
+
+
+type alias Model =
+    { users : RemoteData (List String) (Dict Int User)
+    , underEdit : UnderEdit
+    , errors : List (List String)
+    , confirmingDelete : Maybe ( Int, User )
```

Comment:
Interesting choice (using a product type for the state of the footer). But it kind of makes sense, if after dealing with an Error footer you wanted to go back to an Editing footer. A lot of food for thought for me 🙂 

Were there some alternatives you considered for modelling / the type representation of the application and these "modal" states? Pros/cons? Why was this one the best in the end?

Reply:
Maybe you have misread something? That's not a product type of footer states. Original idea though. Or do you mean the record as a product of named fields?

Reply:
Yeah record type is to me a product type (a way to have multiple items at once - `x AND y AND z` - similarly to tuples etc.), compared to sum types (with custom types / tagged unions / ... you can only have one of the variants - `x OR y OR z`).

You're using `underEdit`, `errors` and `confirmingDelete` to figure out what to show in the footer.

That's where I'm coming from with this question - using product type for the state of the footer. It's very ad-hoc (although that's probably not bad with the scope of this example)

Reply:
Oh ok, in that sense it's a product type indeed.
I felt that it's natural to use a flat model record because underEdit doesn't only affect the footer, but the user elements too, when deciding which one to show the editing UI for. Errors - I didn't give this much thought but I felt that errors also belong in the global scope. At the time of writing this code, I thought of confirmingDelete as the only inherent footer state.